src/OrchardCore.Modules/OrchardCore.Users/Localization/Providers/UserLocalizationRequestCultureProvider.cs

```diff
@@ -0,0 +1,51 @@
+using System;
+using System.Linq;
+using System.Security.Claims;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Localization;
+using Microsoft.Extensions.DependencyInjection;
+using OrchardCore.Localization;
+
+namespace OrchardCore.Users.Localization.Providers;
+
+public class UserLocalizationRequestCultureProvider : RequestCultureProvider
+{
+    public override async Task<ProviderCultureResult> DetermineProviderCultureResult(HttpContext httpContext)
+    {
+        ArgumentNullException.ThrowIfNull(httpContext);
+
+        if (httpContext?.User?.Identity?.IsAuthenticated == false)
+        {
+            return null;
+        }
+
+        string userCulture = null;
+        var claimCulture = httpContext.User.FindFirstValue("culture");
+
+        if (!String.IsNullOrWhiteSpace(claimCulture))
+        {
+            userCulture = claimCulture;
+        }
+
+        if (userCulture == null)
+        {
+            // No values specified for either so no match
+            return null;
+        }
+
+        var localizationService = httpContext.RequestServices.GetService<ILocalizationService>();
+
+        var supportedCulture = await localizationService.GetSupportedCulturesAsync();
+
+        // We verify that the userCulture is still a supportedCulture
+        if (!supportedCulture.Contains(userCulture))
```

Comment:
No need for this, the aim for the request culture provider is to provide the culture, then `RequestLocalizationMiddleware`  will fallback to the default culture if selected culture is not supported

Reply:
I think it is appropriate to return a null ProviderCultureResult here if the user culture doesn't match any supported culture. From my tests, it was relevant when I worked on it.

Reply:
Hmm, here I would agree with @hishamco but need to be tried.

Yes the behavior would be a little different, it will try to fallback to a parent culture (from the previous hyphen), then let another provider win, and then default to the default culture if no supported culture was found in the chain.

Reply:
I might try this later today, I will utilize the time zone difference :)

Reply:
Yes, as said the github search work everywhere unless in the aspnetcore repo for me ;) yes I saw that it's like an helper to use a `Task.FromResult()` when in a method you don't use any `await`.

    NullProviderCultureResult = Task.FromResult(default(ProviderCultureResult));

But when your method already use an `await`, in place of awaiting a `Task.FromResult()`, better to just return `default(ProviderCultureResult)`, so here the type being implied, my suggestion is.

    return default;


 Which returns null because `ProviderCultureResult` is a class (not a value type) so `return null;` was also okay.